python/ray/autoscaler/updater.py

```diff
@@ -74,35 +81,33 @@ def get_node_ip(self):
             return self.provider.external_ip(self.node_id)
 
     def run(self):
-        print("NodeUpdater: Updating {} to {}, logging to {}".format(
-            self.node_id, self.runtime_hash, self.output_name))
+        self.log.info(
+            "NodeUpdater: Updating {} to {}, logging to {}".format(
+                self.node_id, self.runtime_hash, self.output_name))
         try:
             self.do_update()
         except Exception as e:
             error_str = str(e)
             if hasattr(e, "cmd"):
                 error_str = "(Exit Status {}) {}".format(
                     e.returncode, pretty_cmd(" ".join(e.cmd)))
-            print(
-                "NodeUpdater: Error updating {}"
-                "See {} for remote logs.".format(error_str, self.output_name),
-                file=self.stdout)
+            self.log.error("NodeUpdater: Error updating {}"
+                              "See {} for remote logs.".format(
+                                  error_str, self.output_name))
             self.provider.set_node_tags(self.node_id,
                                         {TAG_RAY_NODE_STATUS: "update-failed"})
             if self.logfile is not None:
-                print("----- BEGIN REMOTE LOGS -----\n" +
-                      open(self.logfile.name).read() +
-                      "\n----- END REMOTE LOGS -----")
+                self.log.info("----- BEGIN REMOTE LOGS -----\n" +
```

Comment:
Huh, mutli-line logging messages.

Reply:
This one is kind of weird in that you have to dump the whole thing atomically, otherwise it becomes hard to read with intermingled messages.